### PR TITLE
util: refactor worker with crossbeam-channel

### PR DESCRIPTION
**Pull request by UncP**

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@hicqu @BusyJay PTAL

**Comment by hicqu** [comment]
LGTM.

File: src/util/worker/mod.rs [hunk 1]
```diff
@@ -232,13 +217,11 @@ impl<S: Into<String>> Builder<S> {
 
     pub fn create<T: Display>(self) -> Worker<T> {
         let (scheduler, rx) = if self.pending_capacity == usize::MAX {
-            let (tx, rx) = mpsc::channel::<Option<T>>();
-            let sender = TaskSender::Unbounded(tx);
-            (Scheduler::new(self.name, AtomicUsize::new(0), sender), rx)
+            let (tx, rx) = crossbeam_channel::unbounded::<Option<T>>();
+            (Scheduler::new(self.name, AtomicUsize::new(0), tx), rx)
```
Comment (BusyJay) [inline src/util/worker/mod.rs]: Looks like this line can be put to L226.

File: src/util/worker/mod.rs [hunk 2]
```diff
@@ -158,18 +144,18 @@ impl<T: Display> Scheduler<T> {
     /// If the worker is stopped or number pending tasks exceeds capacity, an error will return.
     pub fn schedule(&self, task: T) -> Result<(), ScheduleError<T>> {
         debug!("scheduling task {}", task);
-        match self.sender {
-            TaskSender::Unbounded(ref tx) => if let Err(SendError(Some(t))) = tx.send(Some(task)) {
+        // whether sender is bounded or not
+        if self.sender.capacity().is_none() {
```
Comment (BusyJay) [inline src/util/worker/mod.rs]: Why not always use `try_send`?

Comment (UncP) [reply src/util/worker/mod.rs]: Maybe we should keep the logic the same as before.

File: src/util/worker/mod.rs (hunk 1 above)
Comment (UncP) [reply src/util/worker/mod.rs]: Will do.

File: src/util/worker/mod.rs (hunk 2 above)
Comment (BusyJay) [reply src/util/worker/mod.rs]: How is it different from before?

Comment (UncP) [reply src/util/worker/mod.rs]: Previously we use `try_send` only when `sender` is bounded.

Comment (BusyJay) [reply src/util/worker/mod.rs]: So what's the motivation behind that?

Comment (UncP) [reply src/util/worker/mod.rs]: I think the error is different, when it is bounded, there are two errors, we should distinguish them. @hicqu

**Comment by UncP** [comment]
should we use `try_send` when it is unbounded? @hicqu

**Comment by siddontang** [comment]
As the official crossbeam channel example shows, it uses try_send for the unbounded channel. So we can use it.

**Comment by hicqu** [comment]
It seems `try_send` is better.  Previously we don't use `try_send` on `unbounded` channel because `UnboundedSender` in `stdlib` doesn't have that method.

**Comment by UncP** [comment]
@BusyJay

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by UncP** [comment]
@BusyJay PTAL

**Pull request merged**